onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -0,0 +1,702 @@
+        ComPtr<IMLOperatorTensorShapeDescription> shapeDesc;
+        context->GetTensorShapeDescription(shapeDesc.GetAddressOf());
+
+        // Get the input and output shape sizes
+        uint32_t inputDimsSize;
+        ORT_THROW_IF_FAILED(shapeDesc->GetInputTensorDimensionCount(0, &inputDimsSize));
+        uint32_t outputDimsSize;
+        ORT_THROW_IF_FAILED(shapeDesc->GetOutputTensorDimensionCount(0, &outputDimsSize));
+        if (inputDimsSize != outputDimsSize)
+        {
+            ORT_THROW_IF_FAILED(E_FAIL);
+        }
+
+        // Get the input shape
+        inputDims_.resize(inputDimsSize);
+        ORT_THROW_IF_FAILED(shapeDesc->GetInputTensorShape(0, static_cast<uint32_t>(inputDims_.size()), inputDims_.data()));
+
+        // Get the output shape
+        outputDims_.resize(outputDimsSize);
+        ORT_THROW_IF_FAILED(shapeDesc->GetOutputTensorShape(0, static_cast<uint32_t>(outputDims_.size()), outputDims_.data()));
+
+        // For the number of total elements in the input and output shapes
+        outputDataSize_ = std::accumulate(outputDims_.begin(), outputDims_.end(), 1, std::multiplies<uint32_t>());
+        inputDataSize_ = std::accumulate(inputDims_.begin(), inputDims_.end(), 1, std::multiplies<uint32_t>());
+
+        // { before_dft_axis, axis, after_dft_axis, real_or_complex }
+        std::array<uint32_t, 4> reshapedInputSize = { 1, 1, 1, inputDims_.back() };
+        std::array<uint32_t, 4> reshapedOutputSize = { 1, 1, 1, outputDims_.back() };
+
+        size_t reshapedIndex = 0;
+        for (int i = 0; i < inputDims_.size() - 1; i++) {
+            if (i == axis_ || i == (axis_ + 1)) {
+                reshapedIndex++;
+            }
+            reshapedInputSize[reshapedIndex] *= inputDims_[i];
+            reshapedOutputSize[reshapedIndex] *= outputDims_[i];
+        }
+
+        auto temporarySize = reshapedInputSize;
+        temporarySize.back() = reshapedOutputSize.back();
+
+        // Calculate elements and strides
+        std::array<uint32_t, 4> reshapedInputStrides = { 1, 1, 1, 1 };
+        std::array<uint32_t, 4> reshapedOutputStrides = { 1, 1, 1, 1 };
+        std::array<uint32_t, 4> temporaryStrides = { 1, 1, 1, 1 };
+        for (int i = static_cast<int>(inputDims_.size()) - 2; i >= 0; i--) {
+            reshapedInputStrides[i] = reshapedInputSize[i + 1] * reshapedInputStrides[i + 1];
+            reshapedOutputStrides[i] = reshapedOutputSize[i + 1] * reshapedOutputStrides[i + 1];
+            temporaryStrides[i] = temporarySize[i + 1] * temporaryStrides[i + 1];
+        }
+
+        // Get DFT Length
+        auto dftLength = inputDims_[axis_];
+
+        // Calculate passes
+        numPasses_ = static_cast<unsigned>(log2(dftLength));
+        bool hasOnePass = numPasses_ == 1;
+        bool hasOddPasses = numPasses_ % 2;
+        bool hasEvenPasses = !hasOddPasses;
+
+        // write directly input buffer to output buffer, dont create temps
+        bool write_to_output = hasOnePass;
```

Comment:
>write_to_output [](http://example.com/codeflow?start=13&length=15)

`writeToOutput` for consistent naming within this folder (currently line 202 isn't even self-consistent within the same function, compared to line 199).